### ISSUE TITLE
Interrested in merging?

### ISSUE THREAD

**Issue by jooooscha** [pta2002/nixvim#35]
Hi, I really like your project, but had missing some plugins so I started to implement some of them in a fork.
On the way I also changed some of your work like some helper functions and for example a lot of the `cmp` module and therefore don't want to create a merge request directly, but wanted to ask you if you even want my work, or if you prefer your way of doing things.
But I feel like at least some of my work could be beneficial to your work.

If you are interested but feel like I changed some things to the worse, I am open for any feedback.
Either way, I will implement more and more modules, and would be happy if other people could benefit from my work as well

**Comment by pta2002** [comment]
Hi! Could you please leave an overview of what's changed, just so I can have an idea of what to look for?

**Comment by jooooscha** [comment]
Sure, 

I think one of the most important change is that I integrated [nixpkgs-vim-extra-plugins](https://github.com/jooooscha/nixpkgs-vim-extra-plugins) into the projekt (Actually a fork of mine, because the developer of the original repo delete plugins as soon they are in nixpkgs, which I think is a bad Idea because they are easier to keep up to date in a separate repo. Adding them is a matter of adding one line to the `manifest.txt` ).
However, I did this to have easy access to more and newer plugins.

I added a mkLuaPlugin helper function. With this functions adding simple lua plugins only requires adding the module options in camalCase and they are automatically converted to the corresponding lua options in snake_case. The function also automatically adds a `enable` option and `extraConfig` option. This also means that new plugins can be added in a matter of minutes, because all lua config can be added via the `extraConfig` options. Nix module options can then be added later.

I used this function to create a list of generated plugins. The Idea is to be able to add new plugins in one line. A complete module can then be added later. This way, we could always have "all' plugins available via modules.

I added a `plugin_template_lua.nix` For easy plugin bootstrapping.

I added aliases for boolOptions, strOptions, intOptions, and their counterparts where the argument can also be null. I added them for cleaner module structure and better overview.

I added a bunch of plugins (besides the generated ones)

As for the cmp module, I modularizes some options as they are deeply nested and I though the module was way too long. I plan to do this for all options eventually.
I started implementing a better sollution for snipped, but did not finish yet. I would revert this before merging, until I finish the new solution.
I think the only breaking change I have made is the way how to add sources. In my implementation they are normal module options that can be enabled and setup via nix; no lua code needed. I think the breaking change would be worth it, but if you really want to avoid breaking anything, I could re-add the old style as an legacy option.

I also renamed a bunch of variables that served the same purpose in different modules but had different names. This is only internally and should not break any config.

I think that is all.
If it is ok I will create a merge request and we can discuss which changes you want and which don't
